### ISSUE TITLE
PublicAPI fixer not working in compiler project

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#66740]
As part of C# 12 work, I need to add some new public APIs to the compiler codebase ([PR to parse roles](https://github.com/dotnet/roslyn/pull/66728)).
But CI fails with `RS0016` ("Symbol is not part of the declared public API", warning treated as error). Locally, I repro that issue with `Build.cmd -runAnalyzers`.

I tried a few of things:
1. going to one of the APIs (in a non-generated source file) and triggering code actions. The spinner bar appears cycling, but even after a long time, I do not get offered a fix to declare the public API. Warning squiggles don't appear either (shown below). I've tried setting the Error List to Build+Intellilsense, then running a Build and running Code Analysis on Solution, but RS0016 didn't appear in the Error List.
2. I tried running the fixer via `dotnet-format` (using our [instructions](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Building%2C%20Debugging%2C%20and%20Testing%20on%20Windows.md#running-the-publicapi-fixer)), but that results in errors shown below. I tried various things (git clean, updating dotnet-format tool) but that didn't help.
3. I tried suppressing RS0016 to unblock this feature work, but that didn't work either. I wasn't sure the proper syntax, so added both `dotnet_public_api_analyzer.RS0016.severity = none` and `dotnet_diagnostic.RS0016.severity = none` to the main editorconfig file, but that had no effect.

![image](https://user-images.githubusercontent.com/12466233/217360401-4037fceb-b631-4f56-89ff-475683302589.png)

![image](https://user-images.githubusercontent.com/12466233/217359738-f5934a2f-7e38-42cc-a51e-c437fe38bed7.png)

FYI @mavasani @jasonmalinowski @sharwell @JoeRobich

**Comment by Youssef1313** [comment]
@jcouv Does it work if you comment out this line?

https://github.com/dotnet/roslyn/blob/a937cb5183b12ff2e042de9d3bb1427312e04271/eng/targets/Imports.targets#L59

**Comment by jasonmalinowski** [comment]
I believe that dotnet format error is something @jmarolf might have on his backlog to investigate.

**Comment by jcouv** [comment]
@Youssef1313 That tip was useful. With it, I was able to get the diagnostics in the IDE. We still have a problem with invoking the fixer on generated files, but I use our standard workaround for that (copy&paste generated source into a source file). I was then able to invoke the fixer to add most syntax APIs.
Somehow, this didn't work for changes in `SyntaxKind.cs`. No clue why, but those are simpler changes, so I got those done by hand.

**Comment by mavasani** [comment]
> Somehow, this didn't work for changes in SyntaxKind.cs. No clue why, but those are simpler changes, so I got those done by hand.

@jcouv I am wondering if this is due to this file being linked into multiple projects, with one of them not having any public API files in the project and hence not having the public API analyzer enabled.

![image](https://user-images.githubusercontent.com/10605811/217453055-e91bf178-bcee-4275-b466-23622ca7fe65.png)

## No diagnostic or code fix
![image](https://user-images.githubusercontent.com/10605811/217452884-fb9df3d5-0bcb-4b65-8a90-af5a9f6e1d7e.png)

## Diagnostic and code fix
![image](https://user-images.githubusercontent.com/10605811/217452737-729a73c6-5e8d-42b4-94d4-8978c0343197.png)

**Comment by jcouv** [comment]
@mavasani The TFM selection explains why I couldn't see the publicAPI fix in `SyntaxKind.cs`. Thanks!
That leaves the dotnet-format question. @jmarolf Can you confirm that is a known issue?

**Comment by jasonmalinowski** [comment]
Assigning this to myself to investigate why the code fix isn't working in the regular generated file; we expect that to work now.

**Comment by jcouv** [comment]
Tried it again today (had to revise the role syntax) and the publicAPI fixer worked. Not sure what happened. Closing until I have a consistent repro. Thanks

**Comment by jcouv** [comment]
Re-opening as the fixer did trigger, but even the solution-wide fix still misses the generated files. Thanks
